src/component.cpp

```diff
@@ -298,8 +316,8 @@ bool Component::addReset(const ResetPtr &reset)
         return false;
     }
 
-    // Prevent adding multiple times to list.
-    if (hasReset(reset)) {
+    // Prevent adding the same reset.
+    if (std::find(mPimpl->mResets.begin(), mPimpl->mResets.end(), reset) != mPimpl->mResets.end()) {
         return false;
     }
```

Comment:
Same comment as above - why are we preventing the addition of duplicate pointers, since every other operation now depends on their contents not their reference pointer?

Reply:
10a2411